tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Rules/MiscellaneousRuleTests.cs

```diff
@@ -141,6 +141,38 @@ public void PropertyDescriptionMustEndWithFullStop(string ruleName, string fullP
         }
     }
 
+    [Theory]
+    [MemberData(nameof(GetAllRules))]
+    public void PropertyNamesMustNotStartWithUnderscore(string ruleName, string fullPath, Type assemblyExporterType)
+    {
+        // Property names that start with underscores are not included in the project cache, so will experience
+        // different behaviour when populated from evaluation vs. cache data. Disallow "private" properties in rules.
+
+        XElement rule = LoadXamlRule(fullPath, out _);
+
+        foreach (var property in GetProperties(rule))
+        {
+            // <Rule>
+            //   <StringProperty Name="Foo">
+
+            string? name = property.Attribute("Name")?.Value;
+
+            if (name is ['_', ..])
```

Comment:
Shouldn't this be failing on https://github.com/dotnet/project-system/blob/f4fe6a4fa35c86fd9b01d3b58c7e1c38d82d7ddf/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/NuGetRestore.xaml#L112 ?

Reply:
See line 162. We skip that one for now. Will tidy up once we can.